nuntium/views.py

```diff
@@ -92,14 +96,21 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class MessageDetailView(DetailView):
-    template_name = 'nuntium/message/message_detail.html'
+class MessageDetailViewMixin(DetailView):
     model = Message
 
     def get_queryset(self):
```

Comment:
That's a good idea - the `get_object` method could include writeitinstance__slug=kwargs['instance_slug'] in it's filters just to make sure that the message is related to the instance that it is being requested from.